# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd 
import numpy as np 

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
s1 = np.random.randint(1,4,size=100)
s2 = np.random.randint(1,3,size=100)
s3 = np.random.randint(10000,30000,size=100)
s1, s2, s3

(array([1, 1, 2, 1, 2, 2, 2, 1, 1, 3, 1, 1, 2, 1, 3, 1, 1, 3, 2, 3, 3, 3,
        2, 2, 1, 3, 3, 3, 2, 3, 2, 1, 3, 2, 3, 1, 3, 2, 2, 1, 1, 2, 3, 2,
        2, 3, 1, 2, 2, 1, 1, 2, 2, 3, 3, 2, 2, 2, 2, 1, 3, 3, 3, 1, 3, 1,
        1, 1, 2, 3, 3, 3, 3, 1, 1, 3, 1, 3, 3, 2, 2, 3, 3, 2, 2, 2, 1, 3,
        3, 3, 2, 1, 2, 2, 2, 1, 2, 3, 3, 3]),
 array([1, 1, 2, 2, 2, 2, 2, 1, 1, 1, 2, 1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 2,
        1, 2, 1, 2, 2, 2, 2, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1,
        1, 2, 2, 2, 2, 1, 2, 2, 1, 2, 1, 2, 1, 1, 2, 2, 2, 1, 1, 1, 2, 1,
        2, 2, 2, 2, 2, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 2,
        2, 2, 1, 2, 2, 2, 2, 1, 2, 1, 2, 1]),
 array([15954, 20561, 17166, 18808, 16576, 26500, 13239, 20517, 17228,
        23758, 16787, 27629, 21976, 27531, 19040, 18569, 23616, 11136,
        26114, 21096, 18682, 15068, 12961, 11258, 25652, 27006, 23480,
        16052, 15701, 16461, 14905, 17787, 23550, 24389, 15380, 15595,
        13275, 18591, 29323, 158

### Step 3. Let's create a DataFrame by joinning the Series by column

In [6]:
df = pd.DataFrame({'s1': s1, 's2': s2, 's3': s3})
df

s1  s2     s3
0    1   1  15954
1    1   1  20561
2    2   2  17166
3    1   2  18808
4    2   2  16576
..  ..  ..    ...
95   1   1  20094
96   2   2  24257
97   3   1  11349
98   3   2  26343
99   3   1  20637

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [8]:
df.columns = ['bedrs', 'bathrs', 'price_sqr_meter']
df

bedrs  bathrs  price_sqr_meter
0       1       1            15954
1       1       1            20561
2       2       2            17166
3       1       2            18808
4       2       2            16576
..    ...     ...              ...
95      1       1            20094
96      2       2            24257
97      3       1            11349
98      3       2            26343
99      3       1            20637

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [11]:
df1 = pd.DataFrame(s1)
df2 = pd.DataFrame(s2)
df3 = pd.DataFrame(s3)
bigcolumn = pd.concat([df1, df2, df3], ignore_index=True)
bigcolumn

0
0        1
1        1
2        2
3        1
4        2
..     ...
295  20094
296  24257
297  11349
298  26343
299  20637

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [13]:
s1 = pd.Series(np.random.randint(1,4,size=100))
s2 = pd.Series(np.random.randint(1,3,size=100))
s3 = pd.Series(np.random.randint(10000,30000,size=100))
s1, s2, s3

(0     1
 1     1
 2     2
 3     2
 4     2
      ..
 95    1
 96    3
 97    1
 98    1
 99    1
 Length: 100, dtype: int32, 0     1
 1     2
 2     2
 3     1
 4     1
      ..
 95    2
 96    1
 97    1
 98    2
 99    1
 Length: 100, dtype: int32, 0     13594
 1     27637
 2     22624
 3     28644
 4     21738
       ...  
 95    26113
 96    10043
 97    19184
 98    28575
 99    17997
 Length: 100, dtype: int32)

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [16]:
df = pd.concat([s1, s2, s3], axis=1)
df

0  1      2
0   1  1  13594
1   1  2  27637
2   2  2  22624
3   2  1  28644
4   2  1  21738
.. .. ..    ...
95  1  2  26113
96  3  1  10043
97  1  1  19184
98  1  2  28575
99  1  1  17997

[100 rows x 3 columns]